### ISSUE TITLE
reproducing issue

### ISSUE THREAD

**Issue by youngwanLEE** [facebookresearch/DiT#21]
Hi, I really impressed by your work and appreciate your clean code. 

I have tried to reproduce DiT-S/8 for 400k steps(about 80 epochs) in Table 4.

In Table. 4, FID-50K score of DiT-S/8 shows `153.6`, but my results are lower than that.

For evaluation, I followed issue [#8](https://github.com/facebookresearch/DiT/issues/8) with some modifications for following Table. 4:
- decoder: vae-mse
- cfg_scale: 0. # because Table 4. denote that the results are based on no cfg.

I have two trials:

1) original  settings([log](https://www.dropbox.com/s/5ejwjso5tklvtdw/DiT-S8-80ep-256batch.txt?dl=0)):
 - `batch size` 256
 - `learning rate` 1e-4
 - `epochs` 80
 - GPUs: 8 x TITAN RTX (24GB)
 
```python
torchrun --nnodes=1 --nproc_per_node=8 train.py --model DiT-S/8 --data-path ~/data/ImageNet/train --num-workers 16 --epochs 80
```

I got the results:

> Inception Score: 6.303833961486816
> FID: 167.51085915419117
> sFID: 65.25114120993317
> Precision: 0.11538
> Recall: 0.0796

2) Increasing batch size settings([log](https://www.dropbox.com/s/ddw93u7zlv80dnt/DiT-S8-80ep-1024batch.txt?dl=0)):
 - `batch size` 1024
 - `learning rate` 1e-4
 - `epochs` 80
 -  - GPUs: 8 x A6000 (48GB)
 
```python
torchrun --nnodes=1 --nproc_per_node=8 train.py --model DiT-S/8 --data-path ~/data/ImageNet/train --num-workers 10 --epochs 80 --global-batch-size 1024
```

I got the results:

> Inception Score: 5.920168876647949
> FID: 177.68215629517465
> sFID: 71.43416470357465
> Precision: 0.1023
> Recall: 0.047

The reason why the bigger batch size one is worse than the first one may be attributed to the `learning rate`.

Is correct to set `cfg_scale` to 0 for without cfg? 

Please let me know if I'm wrong.

I shared my logs.

**Comment by wpeebles** [comment]
Hi @youngwanLEE , you want yo use `cfg_scale=1` to sample without guidance. Using a scale of 0 is (almost) equivalent to sampling from an unconditional diffusion model. You want to keep the scale >= 1 under normal circumstances.

**Comment by heepengpeng** [comment]
have u ever trained  DiT XL_2

**Comment by youngwanLEE** [comment]
@wpeebles 
Thanks for your kind reply. 

I got new result(`FID:154.4`) by following your comment with `cfg_scale=1` . 

The result is close to `153.6`.

> Inception Score: 7.333371162414551
> FID: 154.4886788221333
> sFID: 61.074560135758475
> Precision: 0.1357
> Recall: 0.1041

**Comment by wpeebles** [comment]
Cool, thanks for testing it out @youngwanLEE. By the way, I realized there was a small parameter initialization bug for patchify in `models.py` a couple of days ago and pushed a fix. You might want to make sure your version of the DiT repo is updated with the latest commit before starting other training runs.

**Comment by youngwanLEE** [comment]
@wpeebles Fortunately, I used the updated code. 

I just changed `cfg_scale=1` and the others are not changed. 
I didn't do this 
> Btw, when omitting guidance, you can replace model.forward_with_cfg with model.forward. Just make sure you also don't duplicate your input z, don't concatenate the y_null tokens to your sampled y, and don't delete half of your sampled latents. Sampling should be quite a bit faster if you do this.

Is that correct?

**Comment by youngwanLEE** [comment]
@wpeebles Thanks again!

**Comment by yinhengw** [comment]
Hi @wpeebles @youngwanLEE, in my understanding, `cfg_scale=1` or `cfg_scale=0` makes no difference when sampling (https://github.com/facebookresearch/DiT/blob/main/sample_ddp.py#L81). So why did you get lower FID when `cfg_scale=1`?

In my experiments, I got higher FID with `cfg_scale=0` than `cfg_scale=1.5`. Is it normal, right?

The other question is the results in the following Figure 6 use `cfg_scale=1` or `cfg_scale=0` ?

![Image](https://github.com/user-attachments/assets/f49e8416-824e-428a-9d72-3711e51f77c3)
